### ISSUE TITLE
Crash "undefined class/module VR::SimpleComboBoxEntry (ArgumentError)"

### ISSUE THREAD

**Issue by rubycoder** [Beagle123/visualruby#34]
(I probably have something set up wrong that's causing this, or I'm making some obvious mistake, but I can't find it right now.)

I'm reasonably sure (but not 100% certain) that I had "vr" working correctly on Ubuntu 16.04 (on VirtualBox).  I haven't used (or modified) this Ubuntu installation in about 3 weeks, but I just now tried to use "vr" again.

When I run "vr", the VisualRuby splash screen comes up, then immediately it crashes with an "undefined class/module VR::SimpleComboBoxEntry (ArgumentError)"

I had been using VisualRuby 3.0.20, so I installed the latest 3.0.22, but I get exactly the same problem:

Thanks!!!

```
wayne@ubuntu1604:~$ rvm use ruby-2.3.1
Using /home/wayne/.rvm/gems/ruby-2.3.1
wayne@ubuntu1604:~$ ruby -v
ruby 2.3.1p112 (2016-04-26 revision 54768) [x86_64-linux]
wayne@ubuntu1604:~$ vr
/home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/class_loader.rb:54:in `path2class': undefined class/module VR::SimpleComboBoxEntry (ArgumentError)
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/class_loader.rb:54:in `resolve'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/class_loader.rb:46:in `find'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/class_loader.rb:28:in `load'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/to_ruby.rb:396:in `resolve_class'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/to_ruby.rb:208:in `visit_Psych_Nodes_Mapping'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/visitor.rb:16:in `visit'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/visitor.rb:6:in `accept'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/to_ruby.rb:32:in `accept'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/to_ruby.rb:338:in `block in revive_hash'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/to_ruby.rb:336:in `each'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/to_ruby.rb:336:in `each_slice'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/to_ruby.rb:336:in `revive_hash'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/to_ruby.rb:374:in `revive'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/to_ruby.rb:208:in `visit_Psych_Nodes_Mapping'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/visitor.rb:16:in `visit'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/visitor.rb:6:in `accept'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/to_ruby.rb:32:in `accept'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/to_ruby.rb:311:in `visit_Psych_Nodes_Document'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/visitor.rb:16:in `visit'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/visitor.rb:6:in `accept'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/to_ruby.rb:32:in `accept'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/nodes/node.rb:38:in `to_ruby'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych.rb:253:in `load'
    from /home/wayne/.rvm/gems/ruby-2.3.1/gems/visualruby-3.0.20/lib/SavableClass.rb:13:in `load_yaml'
    from /home/wayne/.rvm/gems/ruby-2.3.1/gems/visualruby-3.0.20/src/main/VR_Main.rb:256:in `load_state'
    from /home/wayne/.rvm/gems/ruby-2.3.1/gems/visualruby-3.0.20/src/main/VR_Main.rb:90:in `load_project'
    from /home/wayne/.rvm/gems/ruby-2.3.1/gems/visualruby-3.0.20/src/main/VR_Main.rb:62:in `before_show'
    from /home/wayne/.rvm/gems/ruby-2.3.1/gems/visualruby-3.0.20/lib/GladeGUI.rb:427:in `show_glade'
    from /home/wayne/.rvm/gems/ruby-2.3.1/gems/visualruby-3.0.20/vr:22:in `load_everything'
    from /home/wayne/.rvm/gems/ruby-2.3.1/gems/visualruby-3.0.20/vr:42:in `block in <top (required)>'
    from /home/wayne/.rvm/gems/ruby-2.3.1/gems/gobject-introspection-3.0.8/lib/gobject-introspection/loader.rb:114:in `invoke'
    from /home/wayne/.rvm/gems/ruby-2.3.1/gems/gobject-introspection-3.0.8/lib/gobject-introspection/loader.rb:114:in `block in define_singleton_method'
    from /home/wayne/.rvm/gems/ruby-2.3.1/gems/visualruby-3.0.20/vr:45:in `<top (required)>'
    from /home/wayne/.rvm/gems/ruby-2.3.1/bin/vr:23:in `load'
    from /home/wayne/.rvm/gems/ruby-2.3.1/bin/vr:23:in `<main>'
    from /home/wayne/.rvm/gems/ruby-2.3.1/bin/ruby_executable_hooks:15:in `eval'
    from /home/wayne/.rvm/gems/ruby-2.3.1/bin/ruby_executable_hooks:15:in `<main>'
wayne@ubuntu1604:~$ gem install visualruby
Fetching: visualruby-3.0.22.gem (100%)
Successfully installed visualruby-3.0.22
Parsing documentation for visualruby-3.0.22
Installing ri documentation for visualruby-3.0.22
Done installing documentation for visualruby after 0 seconds
1 gem installed
wayne@ubuntu1604:~$ vr
/home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/class_loader.rb:54:in `path2class': undefined class/module VR::SimpleComboBoxEntry (ArgumentError)
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/class_loader.rb:54:in `resolve'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/class_loader.rb:46:in `find'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/class_loader.rb:28:in `load'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/to_ruby.rb:396:in `resolve_class'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/to_ruby.rb:208:in `visit_Psych_Nodes_Mapping'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/visitor.rb:16:in `visit'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/visitor.rb:6:in `accept'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/to_ruby.rb:32:in `accept'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/to_ruby.rb:338:in `block in revive_hash'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/to_ruby.rb:336:in `each'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/to_ruby.rb:336:in `each_slice'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/to_ruby.rb:336:in `revive_hash'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/to_ruby.rb:374:in `revive'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/to_ruby.rb:208:in `visit_Psych_Nodes_Mapping'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/visitor.rb:16:in `visit'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/visitor.rb:6:in `accept'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/to_ruby.rb:32:in `accept'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/to_ruby.rb:311:in `visit_Psych_Nodes_Document'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/visitor.rb:16:in `visit'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/visitor.rb:6:in `accept'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/visitors/to_ruby.rb:32:in `accept'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych/nodes/node.rb:38:in `to_ruby'
    from /home/wayne/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/psych.rb:253:in `load'
    from /home/wayne/.rvm/gems/ruby-2.3.1/gems/visualruby-3.0.22/lib/SavableClass.rb:13:in `load_yaml'
    from /home/wayne/.rvm/gems/ruby-2.3.1/gems/visualruby-3.0.22/src/main/VR_Main.rb:266:in `load_state'
    from /home/wayne/.rvm/gems/ruby-2.3.1/gems/visualruby-3.0.22/src/main/VR_Main.rb:90:in `load_project'
    from /home/wayne/.rvm/gems/ruby-2.3.1/gems/visualruby-3.0.22/src/main/VR_Main.rb:62:in `before_show'
    from /home/wayne/.rvm/gems/ruby-2.3.1/gems/visualruby-3.0.22/lib/GladeGUI.rb:427:in `show_glade'
    from /home/wayne/.rvm/gems/ruby-2.3.1/gems/visualruby-3.0.22/vr:20:in `load_everything'
    from /home/wayne/.rvm/gems/ruby-2.3.1/gems/visualruby-3.0.22/vr:40:in `block in <top (required)>'
    from /home/wayne/.rvm/gems/ruby-2.3.1/gems/gobject-introspection-3.0.8/lib/gobject-introspection/loader.rb:114:in `invoke'
    from /home/wayne/.rvm/gems/ruby-2.3.1/gems/gobject-introspection-3.0.8/lib/gobject-introspection/loader.rb:114:in `block in define_singleton_method'
    from /home/wayne/.rvm/gems/ruby-2.3.1/gems/visualruby-3.0.22/vr:43:in `<top (required)>'
    from /home/wayne/.rvm/gems/ruby-2.3.1/bin/vr:23:in `load'
    from /home/wayne/.rvm/gems/ruby-2.3.1/bin/vr:23:in `<main>'
    from /home/wayne/.rvm/gems/ruby-2.3.1/bin/ruby_executable_hooks:15:in `eval'
    from /home/wayne/.rvm/gems/ruby-2.3.1/bin/ruby_executable_hooks:15:in `<main>'
wayne@ubuntu1604:~$ gem list

*** LOCAL GEMS ***

atk (3.0.8)
bigdecimal (1.2.8)
bundler-unload (1.0.2)
cairo (1.15.2)
cairo-gobject (3.0.8)
did_you_mean (1.0.0)
executable-hooks (1.3.2)
gdk3 (3.0.8)
gdk_pixbuf2 (3.0.8)
gem-wrappers (1.2.7)
gio2 (3.0.8)
glib2 (3.0.8)
gobject-introspection (3.0.8)
gtk3 (3.0.8)
gtksourceview3 (3.0.8)
io-console (0.4.5)
json (1.8.3)
minitest (5.8.3)
net-telnet (0.1.1)
pango (3.0.8)
pkg-config (1.1.7)
power_assert (0.2.6)
psych (2.0.17)
rake (10.4.2)
rdoc (4.2.1)
require_all (1.3.3)
rubygems-bundler (1.4.4)
rvm (1.11.3.9)
test-unit (3.1.5)
visualruby (3.0.22, 3.0.20)
wayne@ubuntu1604:~$ which vr
/home/wayne/.rvm/gems/ruby-2.3.1/bin/vr
```

**Comment by Beagle123** [comment]
Hi Wayne:

Somehow you must have restored your old settings file.  Perhaps this is the same settings file you had a problem with last time--the active_record example project.  Just delete the .vr_settings.yaml file, and it will recreate it without using the old SimpleComboBoxEntry class.  Whichever folder vr is opening on startup (likely current folder) the .vr_settings.yaml in that folder is the problem.    Just make sure that you don't restore old settings files.

Best,
Eric

**Comment by Beagle123** [comment]
If you can tell me which project is causing the problem, I can check to make sure I'm not including an old settings file.  I used "grep -r SimpleComboBoxEntry" and got nothing in the while visualruby project.  That means that the text "SimpleComboBoxEntry" doens't appear anywhere on my end.

Maybe you should delete your older vr gem too.

**Comment by rubycoder** [comment]
Thanks for your lightning fast response!!!

What you say makes perfect sense, this problem had a real sense of "deja vu" to me.

I uninstalled both VisualRuby gems, then reinstalled 3.0.22.  Unfortunately, the problem remained exactly the same.

> > Just delete the .vr_settings.yaml file

Unfortunately, since it's been 3 weeks, I'm not at all sure which project I was working on.  I looked through my directories and located a copy of "examples" that I'd made before your fix.  I grepped this directory and found that the problem was indeed in the old active_record example, as you suspected.  Deleting the entire "examples" directory resulted in the "Installing new projects..." dialog coming up at vr startup, and now everything works fine!!!

Could I suggest that to help track down issues in opening projects, it might be worth printing the line
`"Opening project [full path to project]"`
to the console just before opening any project?

Thanks again!!!

Best regards,
Wayne
